deploy/cdk/lib/user-content/user-content-stack.ts

```diff
@@ -116,10 +116,7 @@ export class UserContentStack extends cdk.Stack {
       new CnameRecord(this, `${id}-Route53CnameRecord`, {
         recordName: props.hostnamePrefix,
         domainName: api.domainName!.domainNameAliasDomainName, // cloudfront the api creates
-        zone: HostedZone.fromHostedZoneAttributes(this, 'ApiDomainHostedZone', {
-          hostedZoneId: Fn.importValue(`${props.domainStackName}:Zone`),
-          zoneName: Fn.importValue(`${props.domainStackName}:DomainName`),
-        }),
+        zone: props.foundationStack.hostedZone,
         ttl: Duration.minutes(15),
```

Comment:
```suggestion
        ttl: cdk.Duration.minutes(15),
```

Doing this should allow you to remove L6, I believe.